src/validationinterface.h

```diff
@@ -21,6 +21,7 @@ class CConnman;
 class CValidationInterface;
 class uint256;
 class CScheduler;
+enum class MemPoolRemovalReason;
```

Comment:
I think we can dry-up MemPoolRemovalReason completely in the future. The only control flow pending on it I can see is in `removeUnchecked` and it's only consumed by `TransactionRemovedFromMempool`